packages/account/src/Components/Routes/route-with-sub-routes.jsx

```diff
@@ -18,9 +19,17 @@ const RouteWithSubRoutes = route => {
         } else if (route.is_authenticated && !route.is_logged_in && !route.is_logging_in) {
             redirectToLogin(route.is_logged_in, getLanguage());
         } else {
-            const default_subroute = route.routes ? route.routes.find(r => r.default) : {};
+            let default_subroute = {};
+            route.routes?.forEach(r =>
+                r.subroutes.forEach(subroute => {
+                    if (subroute.default) {
+                        default_subroute = subroute;
+                    }
+                })
+            );
```

Comment:
I am a fan of `reduce` whenever possible but this is just a suggestion. `forEach` is OK too. :smiley: 

```suggestion
            let default_subroute =  (route.routes ?? []).reduce((acc, cur) => ({
                ...acc, 
                ...cur.subroutes.find(subroute => subroute.default)
            }), {})
```

Reply:
Yeah, this looks more legible 🙏🏼